python-sdk/tests/airflow/test_datasets.py

```diff
@@ -0,0 +1,72 @@
+from unittest import mock
+
+import pytest
+
+from astro.airflow.datasets import kwargs_with_datasets
+from astro.sql.table import Table
+
+
+@pytest.mark.parametrize(
+    "kwargs,input_ds,output_ds,dataset_support,expected_kwargs",
+    [
+        (None, None, None, True, {}),
+        ({"task_id": "ex1"}, None, None, True, {"task_id": "ex1"}),
+        (
+            {"task_id": "ex1"},
+            Table("inlet", conn_id="con1"),
+            Table("outlet", conn_id="con2"),
+            True,
+            {
+                "task_id": "ex1",
+                "inlets": [Table("inlet", conn_id="con1")],
+                "outlets": [Table("outlet", conn_id="con2")],
+            },
+        ),
+        (
+            {
+                "task_id": "ex1",
+                "inlets": Table("inlet", conn_id="con1"),
+                "outlets": Table("outlet"),
+            },
+            Table("input_ds"),
+            Table("output_ds"),
+            True,
+            {
+                "task_id": "ex1",
+                "inlets": [Table("inlet", conn_id="con1")],
+                "outlets": [Table("outlet")],
+            },
+        ),
+        # If File is not supported
```

Comment:
If you think it is worth commenting, I would rather add `ids=[...]`. Then we also have proper naming for those tests :)